src/hook/use-navigation-article.js

```diff
@@ -1,33 +1,39 @@
+//useNavigation is available only for Favoris and Search
 import { useContext, useEffect, useState } from 'react';
 import { useNavigate } from 'react-router-dom';
 
 import { ChoicesContext } from '../contexts/choices-context';
 
-//function shared with nav button
-export default function useNavigationArticle({ direction }) {
+export default function useNavigationArticle({ direction, tablNav, pathNav }) {
   const navigate = useNavigate();
-  const { listFavourite, articleChosen, setArticleChosen } = useContext(ChoicesContext);
+  const { articleChosen, setArticleChosen, addArticleToHistory } = useContext(ChoicesContext);
   const [isAvailable, setIsAvailable] = useState(false);
 
-  //found the index of the article where we are
-  const indexActual = listFavourite.findIndex((a) => a.article_id === articleChosen.article_id);
-  const indexAsked = indexActual + direction;
-
-  function handleDirection() {
-    if (isAvailable) {
-      setArticleChosen(listFavourite[indexAsked]);
-      navigate(`/favoris-article-choisi/${listFavourite[indexAsked].article_id}`);
-    }
-  }
+  //find the index of article in display to navigate
+  const indexActuel = tablNav.findIndex((a) => a.article_id === articleChosen.article_id);
+  const indexAsked = indexActuel + direction;
 
+  // check if is first or last article in the tabl to not display component
   useEffect(() => {
-    //control if is the first or last article
-    if (indexAsked < 0 || indexAsked >= listFavourite.length) {
-      setIsAvailable(false);
+    if (tablNav) {
+      if (indexAsked >= 0 && indexAsked < tablNav.length) {
+        setIsAvailable(true);
+      } else {
+        setIsAvailable(false);
+      }
     } else {
-      setIsAvailable(true);
+      setIsAvailable(false);
     }
-  }, [indexActual, indexAsked, listFavourite]);
+  }, [articleChosen, direction, tablNav, indexActuel, indexAsked]);
```

Comment:
Tu peux faire une const simple `isAvailable`
[Pas besoin du useEffect ici](https://react.dev/learn/you-might-not-need-an-effect#updating-state-based-on-props-or-state)